### PR TITLE
Fixes for sklearn 1.7 pre-release support

### PR DESCRIPTION
**Pull request by Alexsandruss**
## Description

Fixes for sklearn 1.7 pre-release support:
- Remove usage of deleted `sklearn.utils._joblib` module in pairwise distances
- Change internal TSNE parameter name to `max_iter` for sklearn>=1.7
- Fix `onedal.utils._array_api._is_numpy_namespace` to support new sklearn's xp name
- Temporary fix probabilities not in [0, 1] interval computed by ensemble models on sklearnex side (sklearn started enforcing upper interval border in estimator check).

Note: this PR fixes critical failures, there are still sklearnex and sklearn test failures.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

N/A.

### CODE REVIEW

File: daal4py/sklearn/ensemble/_forest.py [hunk 1]
```diff
@@ -679,6 +679,12 @@ def _daal_predict_proba(self, X):
         dfc_predictionResult = dfc_algorithm.compute(X, self.daal_model_)
 
         pred = dfc_predictionResult.probabilities
+        # TODO: fix probabilities out of [0, 1] interval on oneDAL side
+        min_pred, max_pred = pred.min(axis=1, keepdims=True), pred.max(
+            axis=1, keepdims=True
+        )
+        pred = (pred - min_pred) / (max_pred - min_pred)
+        pred /= np.sum(pred, axis=1, keepdims=True)
```
Comment (david-cortes-intel) [inline daal4py/sklearn/ensemble/_forest.py:687]: If this happens due to floating point inaccuracies, would be better to just apply a simple clipping.

File: onedal/utils/_array_api.py [hunk 2]
```diff
@@ -68,7 +68,12 @@ def _asarray(data, xp, *args, **kwargs):
 
 def _is_numpy_namespace(xp):
     """Return True if xp is backed by NumPy."""
-    return xp.__name__ in {"numpy", "array_api_compat.numpy", "numpy.array_api"}
+    return xp.__name__ in {
+        "numpy",
+        "array_api_compat.numpy",
+        "numpy.array_api",
+        "sklearn.externals.array_api_compat.numpy",
```
Comment (david-cortes-intel) [inline onedal/utils/_array_api.py:75]: How does one end up with this namespace for an array object? Could it somehow happen that it'd end up with a different name like `<package that imports sklearn>.sklearn.externals.array_api_compat.numpy` ?

**Comment by icfaust** [comment]
Please merge main into the PR and rerun CI for me to review @Alexsandruss

File: onedal/utils/_array_api.py (hunk 2 above)
Comment (icfaust) [reply onedal/utils/_array_api.py:75]: I think you are getting at the core issue here, which is that this function is fragile. We should think about how to make it more robust.

File: daal4py/sklearn/ensemble/_forest.py (hunk 1 above)
Comment (Alexsandruss) [reply daal4py/sklearn/ensemble/_forest.py]: Thanks, clipping actually works.

**Comment by Alexsandruss** [comment]
@mergify backport rls/2025.6.0-rls

**Comment by Alexsandruss** [comment]
@mergify backport rls/2025.5.0-rls
